torch/csrc/jit/codegen/cuda/executor_kernel_arg.h

```diff
@@ -132,22 +134,41 @@ TensorArgAbstract* getTensorArg(int nDims) {
   }
 }
 
-TensorArgAbstract* getTensorArg(c10::ScalarType dtype, int nDims) {
-  switch (dtype) {
-    case (at::kFloat):
-      return getTensorArg<float>(nDims);
-    case (at::kHalf):
-      return getTensorArg<at::Half>(nDims);
-    case (at::kBool):
-      return getTensorArg<bool>(nDims);
-    default:
-      TORCH_CHECK(
-          false,
-          "Dtype: ",
-          dtype,
-          " not currently supported in code generated kernels.");
-  }
-}
+std::unique_ptr<TensorArgAbstract> getTensorArg(
+    c10::ScalarType dtype,
+    int nDims);
+
+class KernelArgumentHolder {
+ public:
+  ~KernelArgumentHolder() = default;
+
+  // Push a tensor to the arguments
+  void push(const at::Tensor& tensor);
+
+  void push(
+      const at::Tensor& tensor,
+      c10::optional<at::IntArrayRef> broadcasted_size);
+
+  // Push a scalar or integer to the arguments
+  void push(const IValue& val);
+
+  void push(const uint64_t& val);
+
+  // Create buffer, flatten arguments into it, align by 8 Bytes, return pointers
+  // in the buffer
+  void** getBuffer();
+
+  void appendArgs(const c10::ArrayRef<c10::IValue>& args);
+
+  void appendArgs(const std::vector<at::Tensor>& tensors);
+
+  void appendPhilox(uint64_t rand_offset);
```

Comment:
add a comment explaining what this does (and what a Philox is?)

Reply:
Who doesn't know about Philox?! Will thoroughly comment when we're closer to usability.